patterns/behavioral/strategy.py

```diff
@@ -7,42 +7,77 @@
 Enables selecting an algorithm at runtime.
 """
 
+from __future__ import annotations
+
+from typing import Callable, Type
+
+
+class DiscountStrategyValidator:  # Descriptor class for check perform
+    def validate(self, obj: Order, value: Callable) -> bool:
```

Comment:
We can make this function a bit more pythonic

Reply:
```suggestion
    @staticmethod
    def validate(obj: Order, value: Callable) -> bool:
        try:
            if obj.price - value(obj) < 0:
                raise ValueError(f"Discount cannot be applied due to negative price resulting. {value.__name__}")
        except ValueError as ex:
            print(str(ex))
        else:
            return True
```

Reply:
```suggestion
    def validate(self, obj: Order, value: Callable) -> bool:
    @staticmethod
    def validate(obj: Order, value: Callable) -> bool:
        try:
            if obj.price - value(obj) < 0:
                raise ValueError(f"Discount cannot be applied due to negative price resulting. {value.__name__}")
        except ValueError as ex:
            print(str(ex))
            return False
        else:
            return True
```

In case not to miss return statement